### PR TITLE
display tree node loading on slow connections

### LINKED ISSUE #3450: Animating background for tree nodes
Now expanding of tree nodes works asynchronously and it is not clear how long it needs to wait until the children appear. It's looks strange when the user click on tree node, arrow on this node has been changed on opened but children appears with some delaying.
Let's see how expanding of tree nodes works with [tree-sample-plugin](https://github.com/eclipse/che-theia-samples/tree/master/tree-view-sample-plugin) provides ftp-explorer widget.

expanding-tree-nodes

In this example the opening time depends on the speed of the network and may take some time.

The proposal is to add animation to the background of the tree node when it's fetching the children.

We did something similar for our Che IDE. Let's see how it looks

animation when expanding tree nodes in che ide

The same video https://youtu.be/RYAopXBC0HU

The styles for the tree node may look like styles for Che IDE

https://github.com/eclipse/che/blob/master/ide/che-core-ide-ui/src/main/resources/org/eclipse/che/ide/ui/smartTree/TreeStyles.css#L144

### LINKED ISSUE #5959: Show progress in Explorer
As a follow-up of https://github.com/theia-ide/theia/pull/5830.

A progress indicator similar to git/scm could be useful.

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- fix #3450
  - mark node as busy after 800ms during refreshing
  - new api `TreeModel.markAsBusy(node, ms, cancellationToken)`  to allow clients indicate that some task is performed for the given node (makes visul effect currently only for composite nodes)
- fix #5959
  - report navigator busy node progress in `explorer` view container
  - new progress location `explorer` for the view container

Copied icons were already were approved in previous CQs.

loading

#### How to test

- emulate slow connection on the system level, for instance with `Network Line Conditioner` or Charles2 proxy
- try to expand some nodes in different trees
  - you should see loading indicator in any tree (i.e. try in npm scripts for the plugin system)
  - you should see blueish bar in the explorer view container for the navigator tree

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/common/progress-service.ts [hunk 1]
```diff
@@ -74,10 +74,7 @@ export class ProgressService {
     async withProgress<T>(text: string, locationId: string, task: () => Promise<T>): Promise<T> {
         const progress = await this.showProgress({ text, options: { cancelable: true, location: locationId } });
         try {
-            const result = task();
```
Comment (akosyakov) [inline packages/core/src/common/progress-service.ts]: @AlexTugarev `await` was missing, so it will immediately cancel progress

Comment (AlexTugarev) [reply packages/core/src/common/progress-service.ts]: thanks!

File: packages/core/src/browser/tree/tree.ts [hunk 2]
```diff
@@ -329,4 +356,29 @@ export class TreeImpl implements Tree {
...
+    async markAsBusy(raw: TreeNode, ms: number, token: CancellationToken): Promise<void> {
+        const node = this.validateNode(raw);
+        if (node) {
+            await this.markAsBusy(node, ms, token);
+        }
+    }
+    protected async doMarkAsBusy(node: Mutable<TreeNode>, ms: number, token: CancellationToken): Promise<void> {
+        try {
+            await timeout(ms, token);
+            this.doSetBusy(node, true);
```
Comment (AlexTugarev) [inline packages/core/src/browser/tree/tree.ts]: exit early here if `token.isCancellationRequested()` is truish, no?

**Comment by AlexTugarev** [comment]
@akosyakov just for completeness, could you please state which CQ you refer to?

File: packages/core/src/browser/tree/tree.ts (hunk 2 above)
Comment (akosyakov) [reply packages/core/src/browser/tree/tree.ts]: yes, it makes sense

**Comment by akosyakov** [comment]
@AlexTugarev https://github.com/eclipse-theia/theia/pull/6993#issuecomment-579777778

File: packages/core/src/browser/style/tree.css [hunk 3]
```diff
@@ -58,28 +58,27 @@
...
 
 .theia-ExpansionToggle {
-    min-width: 10px;
     display: flex;
     justify-content: center;
-    padding-left: calc(var(--theia-ui-padding)*2/3);
-    padding-right: calc(var(--theia-ui-padding)*2/3);
+    padding-left: calc(var(--theia-ui-padding)/2);
+    padding-right: calc(var(--theia-ui-padding)/2);
+    min-width: var(--theia-icon-size);
```
Comment (AlexTugarev) [inline packages/core/src/browser/style/tree.css]: :+1:

File: packages/core/src/browser/tree/tree.ts (hunk 2 above)
Comment (AlexTugarev) [reply packages/core/src/browser/tree/tree.ts]: Ah! Now I got it, it's the `cancelled` exception which will be thrown from `timeout` in that case, thus it won't be set `busy`. 
That wasn't that obvious to me on first scan.

Comment (AlexTugarev) [reply packages/core/src/browser/tree/tree.ts]: maybe a short comment at `timeout`'s def would be fine.

Comment (akosyakov) [reply packages/core/src/browser/tree/tree.ts]: ok, updated docs

**Pull request merged**